xbmc/input/InputManager.cpp

```diff
@@ -394,9 +395,55 @@ bool CInputManager::OnEvent(XBMC_Event& newEvent)
   case XBMC_MOUSEBUTTONDOWN:
   case XBMC_MOUSEBUTTONUP:
   case XBMC_MOUSEMOTION:
-    m_Mouse.HandleEvent(newEvent);
-    ProcessMouse(g_windowManager.GetActiveWindowID());
+  {
+    bool handled = false;
+
+    if (newEvent.type == XBMC_MOUSEMOTION)
+    {
+      static int m_mouseStateX;
+      static int m_mouseStateY;
+      for (std::vector<MOUSE::IMouseHandler*>::iterator it = m_mouseHandlers.begin(); it != m_mouseHandlers.end(); ++it)
+      {
+        if ((*it)->OnMouseMotion(newEvent.motion.x - m_mouseStateX, newEvent.motion.y - m_mouseStateY))
```

Comment:
The `XBMC_Event` struct has fields for relative coordinates, why don't we use those? If we are given relative coords, and libretro requires relative coords, there shouldn't be any need to ever touch absolute positions. This should be possible without any addition or subtraction.